commit 46006c6cf5a8534dc989a3cb7e69f8b11b36893a
Author: Derrick Marcey <dmarcey@github.com>
Date:   Thu May 15 10:44:12 2025 -0700

    Create entrypoint for data route version of page components that calls useRouteQuery instead of useRoutePayload

diff --git a/ui/packages/hosted-compute-ims-stafftools/hosted-compute-ims-stafftools.ts b/ui/packages/hosted-compute-ims-stafftools/hosted-compute-ims-stafftools.ts
index 9b45750653331..959471b5b4cfe 100644
--- a/ui/packages/hosted-compute-ims-stafftools/hosted-compute-ims-stafftools.ts
+++ b/ui/packages/hosted-compute-ims-stafftools/hosted-compute-ims-stafftools.ts
@@ -1,9 +1,15 @@
 import {App} from './App'
-import {MainPageEntrypoint} from './routes/MainPage'
-import {CuratedImageDetailsPageEntrypoint} from './routes/CuratedImageDetailsPage'
+import {MainPageEntrypoint, MainPageEntrypointFuture} from './routes/MainPage'
+import {
+  CuratedImageDetailsPageEntrypoint,
+  CuratedImageDetailsPageEntrypointFuture,
+} from './routes/CuratedImageDetailsPage'
 import {registerDataRouterApp, registerNavigatorApp} from '@github-ui/react-core/register-app'
 import {jsonRoute} from '@github-ui/react-core/json-route'
-import {CuratedImageVersionDetailsPageEntrypoint} from './routes/CuratedImageVersionDetailsPage'
+import {
+  CuratedImageVersionDetailsPageEntrypoint,
+  CuratedImageVersionDetailsPageEntrypointFuture,
+} from './routes/CuratedImageVersionDetailsPage'
 import {hostedComputeImsStafftoolsAppBuilder} from './config/app-builder'
 import {hostedComputeImageDetailsRoute} from './routes/hosted-compute-image-details-route'
 import {hostedComputeImageVersionDetailsRoute} from './routes/hosted-compute-image-version-details-route'
@@ -26,13 +32,13 @@ registerNavigatorApp('hosted-compute-ims-stafftools', () => ({

 export const hostedComputeImsStafftoolsApp = hostedComputeImsStafftoolsAppBuilder.createDataRouterAppFromRoutes([
   hostedComputeImsAdminRoute.toRoute({
-    Component: MainPageEntrypoint,
+    Component: MainPageEntrypointFuture,
   }),
   hostedComputeImageDetailsRoute.toRoute({
-    Component: CuratedImageDetailsPageEntrypoint,
+    Component: CuratedImageDetailsPageEntrypointFuture,
   }),
   hostedComputeImageVersionDetailsRoute.toRoute({
-    Component: CuratedImageVersionDetailsPageEntrypoint,
+    Component: CuratedImageVersionDetailsPageEntrypointFuture,
   }),
 ])

diff --git a/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageDetailsPage.tsx b/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageDetailsPage.tsx
index fa2b53ed52814..d83276e73e90b 100644
--- a/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageDetailsPage.tsx
+++ b/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageDetailsPage.tsx
@@ -6,6 +6,8 @@ import {PageHeader} from '@primer/react'
 import {pageHeadingStyle} from '../helpers/style'
 import {CuratedImageDetailsView} from '../components/CuratedImageDetailsView'
 import {useEffect} from 'react'
+import {useRouteQuery} from '@github-ui/react-core/future/use-route-query'
+import {hostedComputeImageDetailsRoute} from './hosted-compute-image-details-route'

 export interface CuratedImageDetailsPagePayload {
   imageDefinition: ImageDefinition
@@ -18,6 +20,11 @@ export function CuratedImageDetailsPageEntrypoint() {
   return <CuratedImageDetailsPage payload={payload} />
 }

+export function CuratedImageDetailsPageEntrypointFuture() {
+  const {data: payload} = useRouteQuery(hostedComputeImageDetailsRoute, 'mainQuery')
+  return <CuratedImageDetailsPage payload={payload} />
+}
+
 function CuratedImageDetailsPage({payload}: {payload: CuratedImageDetailsPagePayload}) {
   useEffect(() => {
     window.scrollTo(0, 0)
diff --git a/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageVersionDetailsPage.tsx b/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageVersionDetailsPage.tsx
index ed5e6d8388799..41048719f7f29 100644
--- a/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageVersionDetailsPage.tsx
+++ b/ui/packages/hosted-compute-ims-stafftools/routes/CuratedImageVersionDetailsPage.tsx
@@ -6,6 +6,8 @@ import {PageHeader} from '@primer/react'
 import {pageHeadingStyle} from '../helpers/style'
 import {CuratedImageVersionHeaderDescription} from '../components/CuratedImageVersionHeaderDescription'
 import {useEffect} from 'react'
+import {hostedComputeImageVersionDetailsRoute} from './hosted-compute-image-version-details-route'
+import {useRouteQuery} from '@github-ui/react-core/future/use-route-query'

 export interface CuratedImageVersionDetailsPagePayload {
   imageDefinition: ImageDefinition
@@ -18,6 +20,11 @@ export function CuratedImageVersionDetailsPageEntrypoint() {
   return <CuratedImageVersionDetailsPage payload={payload} />
 }

+export function CuratedImageVersionDetailsPageEntrypointFuture() {
+  const {data: payload} = useRouteQuery(hostedComputeImageVersionDetailsRoute, 'mainQuery')
+  return <CuratedImageVersionDetailsPage payload={payload} />
+}
+
 function CuratedImageVersionDetailsPage({payload}: {payload: CuratedImageVersionDetailsPagePayload}) {
   useEffect(() => {
     window.scrollTo(0, 0)
diff --git a/ui/packages/hosted-compute-ims-stafftools/routes/MainPage.tsx b/ui/packages/hosted-compute-ims-stafftools/routes/MainPage.tsx
index a47ff2933d041..ca6e5d52e12c2 100644
--- a/ui/packages/hosted-compute-ims-stafftools/routes/MainPage.tsx
+++ b/ui/packages/hosted-compute-ims-stafftools/routes/MainPage.tsx
@@ -9,6 +9,8 @@ import {changeUrlParam} from '../helpers/utils'
 import type {SelectedImagesTab} from '../helpers/urls'
 import {PageHeader} from '@primer/react'
 import {pageHeadingStyle} from '../helpers/style'
+import {useRouteQuery} from '@github-ui/react-core/future/use-route-query'
+import {hostedComputeImsAdminRoute} from './hosted-compute-ims-admin-route'

 export interface MainPagePayload {
   imageDefinitions: ImageDefinition[]
@@ -20,6 +22,11 @@ export function MainPageEntrypoint() {
   return <MainPage payload={payload} />
 }

+export function MainPageEntrypointFuture() {
+  const {data: payload} = useRouteQuery(hostedComputeImsAdminRoute, 'mainQuery')
+  return <MainPage payload={payload} />
+}
+
 function MainPage({payload}: {payload: MainPagePayload}) {
   const [selectedTab, setSelectedTab] = useState<SelectedImagesTab>(payload.selectedImagesTab || 'github-images')
+


